hatch/src/assets/generator.rs

```diff
@@ -0,0 +1,26 @@
+use assets::{ Asset, ProjectAsset };
+use hatch_error::{ HatchResult, ResultExt };
+use std::fs;
+use std::io::prelude::*;
+use std::io::Write;
+
+pub fn generate_one(asset: &ProjectAsset) -> HatchResult<()> {
+  let path = asset.path();
+  fs::create_dir_all(&path)?;
```

Comment:
Can we wrap these with better error messages?

```rust
fs::create_dir_all(&path).with_context(|e| {
  format!("Failed to create directory: `{}` : {}", &path, e)
})?;
```

Reply:
Absolutely!